src/test/kotlin/com/igorwojda/integer/generateallpairs/challenge.kt

```diff
@@ -4,7 +4,14 @@ import org.amshove.kluent.shouldBeEqualTo
 import org.junit.jupiter.api.Test
 
 private fun getAllPairs(n: Int): List<Pair<Int, Int>> {
-    TODO("not implemented")
+    //return List((n+1)*(n+1)){i -> Pair(i,i)}
+    val listOfPairs = ArrayList<Pair<Int, Int>>()
+    for(i in 0..n){
+        for(j in 0..n){
+            listOfPairs.add(Pair(i,j))
```

Comment:
Can you think of an infix function (https://kotlinlang.org/docs/reference/functions.html#infix-notation) that can make your code a lil bit more readable?

Reply:
well?

Reply:
For some reason, the edited version is showing on my local branch but not here

Reply:
Now it's showing in my last commit 😄